flatdata-rs/lib/src/tarstorage.rs

```diff
@@ -0,0 +1,128 @@
+use crate::storage::{ResourceStorage, StorageHandle, Stream};
+
+use memmap2::Mmap;
+use std::{
+    collections::HashMap,
+    fs::File,
+    io,
+    ops::Range,
+    path::{Path, PathBuf},
+    slice,
+    sync::Arc,
+};
+
+/// Internal storage of file entries in tar archive.
+#[derive(Debug)]
+struct MemoryMappedTarArchiveStorage {
+    archive_map: Mmap,
+    file_ranges: HashMap<PathBuf, Range<usize>>,
+}
+
+impl MemoryMappedTarArchiveStorage {
+    pub fn new(tar_path: &Path) -> Result<Self, io::Error> {
+        let file = File::open(tar_path)?;
+        let mut archive = tar::Archive::new(file);
```

Comment:
Do you have to pass the file here, or can you also pass the memory mapped file? (I think you can convert a byte slice into Read/whatever trait is needed here, e.g. using `std::io::Cursor`)